src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs

```diff
@@ -48,12 +47,43 @@ void Report(Scope current)
                     }
 
                     callback(activityLogScope, state);
+
+                    // Tags and baggage are opt-in and thus we assume that most of the time it will not be used.
+                    if ((_activityTrackingOption & ActivityTrackingOptions.Tags) != 0
+                        && activity.TagObjects.GetEnumerator().MoveNext())
+                    {
+                        // As TagObjects is a IEnumerable<KeyValuePair<string, object?>> this can be used directly as a scope.
+                        // We do this to safe the allocation of a wrapper object.
+                         callback(activity.TagObjects, state);
+                    }
+
+                    if ((_activityTrackingOption & ActivityTrackingOptions.Baggage) != 0
```

Comment:
I believe every time you call activity.Baggage will allocate a new object. would be good if you can do 

```C#
IEnumerable<KeyValuePair<string, string?>> baggage = activity.Baggage;
if (baggage.GenEnumerator().MoveNext())
{
    AdditionalActivityItemsLogScope<string?> scope = LogScopeForAdditionalActivityItems(activity, baggage);
    callback(scope, state);
}
```